src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -170,35 +166,56 @@ namespace AppInstaller::Repository::Microsoft::Schema::V1_4
     {
         using namespace SQLite::Builder;
         SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_4");
+        constexpr std::string_view dependencyIndexByManifestId = "dependencies_manifest_id_index";
+        constexpr std::string_view dependencyIndexByPackageId = "dependencies_package_id_index";
 
         StatementBuilder createTableBuilder;
         createTableBuilder.CreateTable(TableName()).BeginColumns();
         createTableBuilder.Column(IntegerPrimaryKey());
 
-        std::vector<DependenciesTableColumnInfo> dependenciesColumns =
+        std::vector<DependenciesTableColumnInfo> notNullableDependenciesColumns =
         {
             { s_DependenciesTable_Manifest_Column_Name },
             { s_DependenciesTable_MinVersion_Column_Name },
+        };
+
+        std::vector<DependenciesTableColumnInfo> nullableDependenciesColumns =
+        {
             { s_DependenciesTable_PackageId_Column_Name }
         };
 
-        for (const DependenciesTableColumnInfo& value : dependenciesColumns)
+        for (const DependenciesTableColumnInfo& value : notNullableDependenciesColumns)
         {
             createTableBuilder.Column(ColumnBuilder(value.Name, Type::RowId).NotNull());
         }
 
+        for (const DependenciesTableColumnInfo& value : nullableDependenciesColumns)
+        {
+            createTableBuilder.Column(ColumnBuilder(value.Name, Type::RowId));
+        }
+
         createTableBuilder.EndColumns();
 
         createTableBuilder.Execute(connection);
 
+        // Primary key index by package rowid and manifest rowid.
         StatementBuilder createPKIndexBuilder;
-        createPKIndexBuilder.CreateUniqueIndex(s_DependenciesTable_Index_Name).On(s_DependenciesTable_Table_Name).Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name });
+        createPKIndexBuilder.CreateUniqueIndex(s_DependenciesTable_Index_Name).On(s_DependenciesTable_Table_Name).Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_PackageId_Column_Name });
         createPKIndexBuilder.Execute(connection);
 
+        // Index of dependency by Manifest id. 
+        StatementBuilder createIndexByManifestIdBuilder;
+        createIndexByManifestIdBuilder.CreateIndex(dependencyIndexByManifestId).On(s_DependenciesTable_Table_Name).Columns({ s_DependenciesTable_Manifest_Column_Name });
```

Comment:
Since the manifest is the first column on the primary key index, this one isn't helping any.  But you probably do want one on the version to make the "is referenced" function faster.